lib/Doctrine/DBAL/Driver/IBMDB2/DB2Statement.php

```diff
@@ -199,9 +199,9 @@ public function setFetchMode($fetchMode, $arg2 = null, $arg3 = null)
      */
     public function getIterator()
     {
-        $data = $this->fetchAll();
-
-        return new \ArrayIterator($data);
+        while ($row = $this->fetch()) {
```

Comment:
What if the fetch mode is `PDO::FETCH_COLUMN` and one of the values is empty? Probably should strictly compare with `FALSE` and add this to the test.